### PR TITLE
Edit/delete form btn

### PR DESCRIPTION
**Pull request by kehngid**
Hey, sorry it looks like so many changes. My extension that will reformat the code messes with the tab size so it looks like everything is changed. 

The big changes are: 
- Delete Modal
- Two new buttons on card listing that are visible depending on user id and role. 
- Edit Modal which is just a CreateListingModal that will appear pre-populated when you click edit button

### CODE REVIEW

**Comment by kehngid** [comment]
Need to modify initial values in CreateListing Model so it works with new Listing Schema

**Comment by kehngid** [comment]
Should be noted I'll need to go into CreateListingModal and correct the initialListingsValues in the if stmt, should look like this: bathrooms: props.data?.bathrooms || '' for all of the Lisiting attributes, I just forgot about it while I was doing conflict resolutions

File: src/components/listings/CreateListingModal.tsx [hunk 1]
```diff
@@ -84,34 +91,53 @@ function CreateListingModal() {
...
-        console.log(values);
-        const response = await fetch('https://jsonplaceholder.typicode.com/posts/1')
-        const data = await response.json();
-        console.log(data);
-        setShow(false);
-    }
+  const formRef = useRef<FormikProps<CreateListingSchema> | null>(); //| null
+  const [show, setShow] = useState(false);
+  const handleFormSubmit = async (values: any) => {
+    console.log("Values: " + values);
```
Comment (TheRyanMills) [inline src/components/listings/CreateListingModal.tsx]: Log!

File: src/components/listings/CreateListingModal.tsx [hunk 2]
```diff
@@ -84,34 +91,53 @@ function CreateListingModal() {
...
+    const response = await fetch(
+      //"https://jsonplaceholder.typicode.com/posts/1"
+      "http://localhost:3000/:id/save" 
+      );
+    const data = await response.json();
+    console.log(data);
```
Comment (TheRyanMills) [inline src/components/listings/CreateListingModal.tsx]: Log!

File: src/components/listings/CreateListingModal.tsx [hunk 3]
```diff
@@ -84,34 +91,53 @@ function CreateListingModal() {
...
-                    <Modal.Title>Create New Listing</Modal.Title>
-                </Modal.Header>
-                <Modal.Body>
-                    <Formik
-                        innerRef={(f) => (formRef.current = f)}
-                        validationSchema={schema}
-                        onSubmit={handleFormSubmit}
-                        initialValues={{
+  if ("data" in props) {
+    console.log("yay data is here");
```
Comment (TheRyanMills) [inline src/components/listings/CreateListingModal.tsx]: Log!

File: src/components/listings/CreateListingModal.tsx [hunk 4]
```diff
@@ -140,7 +166,25 @@ function CreateListingModal() {
                             quietHoursEnd: "",
                             startDate: "",
                             endDate: ""
-                        }}
+                        };
+  } else {
+    console.log("Aw no data");
```
Comment (TheRyanMills) [inline src/components/listings/CreateListingModal.tsx]: Log!

File: src/pages/BrowsePage.tsx [hunk 5]
```diff
@@ -1,46 +1,43 @@
...
+      is_private_room: "yes",
+      move_in_date: "1/27/2024",
+      move_out_date: "9/26/2024",
+    },
+  ];
+
+  useEffect(() => {
+    FetchData("http://localhost:3000/listings").then(([data]) => {
+      setData(data);
+      console.log(fetchedData);
```
Comment (TheRyanMills) [inline src/pages/BrowsePage.tsx]: Log!

File: src/components/listings/CreateListingModal.tsx [hunk 6]
```diff
@@ -84,34 +91,53 @@ function CreateListingModal() {
...
-        const data = await response.json();
-        console.log(data);
-        setShow(false);
-    }
+  const formRef = useRef<FormikProps<CreateListingSchema> | null>(); //| null
+  const [show, setShow] = useState(false);
+  const handleFormSubmit = async (values: any) => {
+    console.log("Values: " + values);
+    const response = await fetch(
+      //"https://jsonplaceholder.typicode.com/posts/1"
```
Comment (TheRyanMills) [inline src/components/listings/CreateListingModal.tsx]: Comment!

File: src/components/listings/ListingCard.tsx [hunk 7]
```diff
@@ -1,23 +1,40 @@
...
-	const [isCheckoutModalOpen, setIsCheckoutModalOpen] = useState(false);
-	//const picID = Number(props._id.substring(0,2));
-	const picID = props.rent % 1000;
+  const [imgLoading, setImgLoading] = useState(true);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isCheckoutModalOpen, setIsCheckoutModalOpen] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [isCreateListingModelOpen, setCreateListingModalOpen] = useState(false);
+  const [permissions, setPermissions] = useState(false);
+  //const picID = Number(props._id.substring(0,2));
```
Comment (TheRyanMills) [inline src/components/listings/ListingCard.tsx]: Comment!

File: src/pages/BrowsePage.tsx [hunk 8]
```diff
@@ -1,46 +1,43 @@
...
-		FetchData('http://localhost:3000/listings')
-		.then(([data]) => {
-			setData(data);
-			//console.log(fetchedData);
-		});
-	  }, []);
-
-	return <ListingCardGrid dataList={fetchedData} />;
+  const [fetchedData, setData] = useState<ListingCardData[]>([]);
+  //const { isLoggedIn, role } = useContext(UserContext);
```
Comment (TheRyanMills) [inline src/pages/BrowsePage.tsx]: Comment!

**Review by TheRyanMills — CHANGES_REQUESTED** [review]
Marked some comments and console logs that slipped through.

**Pull request merged**